KyberElasticAdapter/contracts/UniswapV3likeAdapter.sol

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "./interfaces/IERC20.sol";
+import "./lib/SafeERC20.sol";
+import "./BaseAdapter.sol";
+
+struct QParams {
+    address tokenIn;
+    address tokenOut;
+    int256 amountIn;
+    uint24 fee;
+}
+
+interface IUniV3Pool {
+    function swap(
+        address recipient,
+        bool zeroForOne,
+        int256 amountSpecified,
+        uint160 sqrtPriceLimitX96,
+        bytes calldata data
+    ) external returns (int256 amount0, int256 amount1);
+
+    function token0() external view returns (address);
+
+    function token1() external view returns (address);
+
+    function liquidity() external view returns (uint128);
+}
+
+interface IUniV3Quoter {
+    function quoteExactInputSingle(QParams memory params)
+        external
+        view
+        returns (uint256);
+
+    function quote(
+        address,
+        bool,
+        int256,
+        uint160
+    ) external view returns (int256, int256);
+}
+
+abstract contract UniswapV3likeAdapter is BaseAdapter {
+    using SafeERC20 for IERC20;
+
+    uint160 internal constant MAX_SQRT_RATIO =
+        1461446703485210103287273052203988822378723970342;
+    uint160 internal constant MIN_SQRT_RATIO = 4295128739;
+    uint256 public quoterGasLimit;
+    address public quoter;
+
+    constructor(
```

Comment:
Missing NatSpec
Add descriptions for the interface, events and functions
Refer https://docs.soliditylang.org/en/v0.8.17/natspec-format.html